pages/k8s/upgrade-notes.md

```diff
@@ -14,12 +14,162 @@ toc: False
+
+juju deploy cs:~containers/kubernetes-worker-$CURRENT_WORKER_REV  kubernetes-worker-temp -n $CURRENT_WORKER_COUNT
+```
+
+#### Add necessary relations
+```bash
+juju status --relations | grep worker: | awk '{print $1,$2}' | sed 's/worker:/worker-temp:/g' | xargs -n2 juju relate
+```
+Wait for the temporary workers to become active before continuing.
+Upgrade the master and worker charms:
+```bash
+juju upgrade-charm kubernetes-master
+juju upgrade-charm kubernetes-worker
+```
+
+The kubernetes-worker units will enter a blocked state, with status message
+“Connect a container runtime.”
+
+#### Deploy and relate the new Docker charm
+
+This step is needed even if you do not intend to use Docker following the
+upgrade. Docker is already installed on your `kubernetes-worker` units, and
+the Docker subordinate includes clean-up code to uninstall Docker when the
+Docker charm is replaced with the containerd charm.
+
+```bash
+juju deploy docker
+juju add-relation docker kubernetes-master
+juju add-relation docker kubernetes-worker
+```
+(Currently juju deploy cs:~joeborg/docker-0)
+
+### Switching to Containerd
+
+Now, pause the existing workers, which will move pods to the temporary workers.
+
+```bash
+juju run-action [unit] pause --wait
+```
+
+For example:
+```bash
+juju run-action kubernetes-worker/0 pause --wait
+juju run-action kubernetes-worker/1 pause --wait
+juju run-action kubernetes-worker/2 pause --wait
+```
+
+One-liner:
+```bash
+juju status | grep ^kubernetes-worker/ | awk '{print $1}' | tr -d '*' | xargs -n1 -I '{}' juju run-action {} pause --wait
+```
+
+#### Remove Docker
+
+This will uninstall Docker from the workers.
+
+```bash
+juju remove-relation docker kubernetes-master
+juju remove-relation docker kubernetes-worker
+juju remove-application docker
+```
+
+#### Deploy Containerd
+
+```bash
+juju deploy containerd
+juju add-relation containerd kubernetes-master
+juju add-relation containerd kubernetes-worker
+```
+
+#### Resume workers
+Now we can allow pods to be rescheduled to our original workers.
+
+```bash
+juju run-action [unit] resume --wait
+```
+
+E.g.
+```bash
+juju run-action kubernetes-worker/0 resume --wait
+juju run-action kubernetes-worker/1 resume --wait
+juju run-action kubernetes-worker/2 resume --wait
+```
+
+One-liner:
+```bash
+juju status | grep ^kubernetes-worker/ | awk '{print $1}' | tr -d '*' | xargs -n1 -I '{}' juju run-action {} resume --wait
+```
+
+#### Cleanup
+
+You can now pause the temporary workers to force all pods to migrate back
+to your “real” workers, then remove the temporary workers.
+
+```bash
+juju status | grep ^kubernetes-worker-temp/ | awk '{print $1}' | tr -d '*' | xargs -n1 -I '{}' juju run-action {} pause --wait
+
+juju remove-application kubernetes-worker-temp
+```
+
+
+#### Mixing Containerd and Docker
+
+
+Once you have a Containerd backed CDK running, you can add Docker backed
+workers like so:
+
+```bash
+juju deploy cs:~containers/kubernetes-worker-docker
```

Comment:
@joedborg Is this right? There is a different charm for docker based workers? Because we are using the same one above (line 44)

Reply:
I mean, I would understand deploying it **as** a different name

Reply:
I would expect this line to be

```
juju deploy cs:~containers/kubernetes-worker kubernetes-worker-docker
```

Same charm, different application name.

Reply:
It's meant be an explicit name, e.g. ` juju deploy cs:~containers/kubernetes-worker kubernetes-worker-docker`

Reply:
okay, I will fix it here and on the container page